cms/djangoapps/contentstore/rest_api/v0/views/advanced_settings.py

```diff
@@ -0,0 +1,152 @@
+""" API Views for course advanced settings """
+
+import edx_api_doc_tools as apidocs
+from opaque_keys.edx.keys import CourseKey
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from xmodule.modulestore.django import modulestore
+
+from cms.djangoapps.models.settings.course_metadata import CourseMetadata
+from common.djangoapps.student.auth import has_studio_read_access, has_studio_write_access
+from openedx.core.lib.api.view_utils import DeveloperErrorViewMixin, verify_course_exists, view_auth_classes
+from ..serializers import CourseAdvancedSettingsSerializer
+from ....views.course import update_course_advanced_settings
+
+
+@view_auth_classes(is_authenticated=True)
+class AdvancedCourseSettingsView(DeveloperErrorViewMixin, APIView):
+    """
+    View for getting and setting the advanced settings for a course.
+    """
+
+    @apidocs.schema(
+        parameters=[apidocs.string_parameter("course_id", apidocs.ParameterLocation.PATH, description="Course ID")],
+        responses={
+            200: CourseAdvancedSettingsSerializer,
+            401: "The requester is not authenticated.",
+            403: "The requester cannot access the specified course.",
+            404: "The requested course does not exist.",
+        },
+    )
+    @verify_course_exists()
+    def get(self, request: Request, course_id: str):
+        """
+        Get an object containing all the advanced settings in a course.
```

Comment:
@xitij2000 Do we need a list of all advanced settings in the course? For the discussions app we only need the settings related to discussions. How do you plan to use this API in the MFE?

Reply:
@asadazam93  This isn't about discussions, it is about pages and resources, so settings related to progress, wiki etc. 

We don't need all the settings, but I felt it better to expose the existing generic API that can be used for multiple purposes, than creating a very constricted API with no clear theme for a specific purpose.